src/main/java/com/github/tomakehurst/wiremock/matching/AbstractDateTimeMatchResult.java

```diff
@@ -83,7 +83,7 @@ public double getDistance() {
   }
 
   private double calculateDistance(Temporal start, Temporal end) {
-    double distance = ((double) ChronoUnit.YEARS.between(start, end)) / 100;
+    double distance = ((double) ChronoUnit.MINUTES.between(start, end)) / (365 * 24 * 60 * 100);
```

Comment:
Any reason we couldn't go down to milliseconds? e.g.

```java
private static final long ONE_YEAR_IN_MILLIS = 365 * 24 * 60 * 60 * 1000L;

  private double calculateDistance(Temporal start, Temporal end) {
    double distance = ((double) (ChronoUnit.MILLIS.between(start, end))) / ONE_YEAR_IN_MILLIS;
    distance = Math.abs(distance);
    return Math.min(distance, 1.0);
  }
```

Reply:
I tried going down to milliseconds and hit a numeric overflow not realizing I should use the `L` suffix so that the denominator is calculated using longs instead of integers. My mistake, I will fix that on the next commit.

Before making the change I will need clarification: Right now all date/time distances are normalized within the range from 0 to 1. In your initial implementation you set 100 years as a ceiling when calculating date/time distances. That is, any distance greater than 100 years results in a distance of 1.

In your suggestion above you lower that ceiling even further to one year. Any date/time distance greater than a year will now result in a distance of one. I suspect that was by mistake and we should stick to the 100 year ceiling, right?

On the other hand, could we even raise that ceiling instead? Is there a reason 100 years should be the limit over which all distances equal to 1?

Reply:
No I did it deliberately, but this approach is pretty arbitrary and I'm not really happy it in general TBH.

The issue is that normalising to a finite range is hard when the potential difference in date values in infinite.

I think that the difference in dates that are already quite close are more significant (people will probably mostly be working in the range of few days-months at most from today) so a better strategy would probably be a logarithmic function asymptotic to 1.0. If you felt like tackling that also, that would be a significant improvement.

Reply:
I have amended my commit, this time using a version of the following function:
```
          |x|
f(x) = --------
       |x| + c
```
When c > 0, f has a range of [0, 1) and is asymptotic to 1 as shown in the following graph (where c = 1):
![graph](https://github.com/wiremock/wiremock/assets/4935979/5ee5368e-7f30-4f59-94f9-a383a5435434)

In our case the actual `calculateDistance` function implemented has the form:
```
                 |time diff|
distance = -----------------------
           |time diff| + 50 years
```
The number 50 is arbitrary but it keeps to the previous implementation of having 100 years as the ceiling. That is, a 50 year difference still results in a 0.5 distance.

I have edited the title of the PR to reflect its new scope.

Please let me know what you think.